lib/ReactViews/Custom/ChartCustomComponent.ts

```diff
@@ -224,7 +224,15 @@ export default abstract class ChartCustomComponent<
       // Build expand/download buttons
       const sourceItems = (attrs.downloads || attrs.sources || [""]).map(
         (source: string, i: number) => {
-          const id = `${context.catalogItem.uniqueId}:${source}`;
+          // When expanding a chart for this item and there is already an
+          // expanded chart for the item, there are 2 possibilities.
+          // 1. Remove it an show the new chart
+          // 2. Show the new chart alongside the existing chart
+          //
+          // If title & source names for the two expanded charts are the same then
+          // we only show the latest one, otherwise we show both.
+          // To do this we make the id dependant on the parentId, title & source.
+          const id = `${context.catalogItem.uniqueId}:${attrs.title}:${source}`;
```

Comment:
is there an example dataset where I can see this?

Reply:
- Example link for behavior 2 i.e when expanding chart from a catalog item and chart title is different from the currently expanded one, create a new chart item - [link](http://ci.terria.io/chart-fixes/#share=s-dShjXJ4zG9AUnTm2KAOHFFQVZ09&https://gist.githubusercontent.com/na9da/3c46309baf6cca7be9d301e6348eb75e/raw/e35b8ef8d21d9016a3ec8f23d0df5ab56d940532/coastal.json)
- Example link for behavior 1 i.e when expanding chart from a catalog item and chart title is the same, replace the currently expanded chart for the item with the new chart. I couldn't locate another example dataset for this, but I'm sure it exists :-)